iam-login-service/src/test/java/it/infn/mw/iam/test/oauth/profile/ScopeAwareProfileResolverTests.java

```diff
@@ -118,5 +118,57 @@ public void multipleProfilesLeadToDefaultProfile() {
     profile = profileResolver.resolveProfile(Set.of("openid", "kc", "wlcg"));
     assertThat(profile, is(iamProfile));
 
+    profile = profileResolver.resolveProfile(Set.of());
+    assertThat(profile, is(iamProfile));
   }
+
+  @Test
+  public void multipleProfilesLeadToRequestedProfile() {
+
+    Set<String> clientScopes = Set.of("openid", "aarc", "wlcg");
+
+    JWTProfile profile = profileResolver.resolveProfile(clientScopes, Set.of("openid", "wlcg"));
+    assertThat(profile, is(wlcgProfile));
+
+    profile = profileResolver.resolveProfile(clientScopes, Set.of("openid", "aarc"));
+    assertThat(profile, is(aarcProfile));
+
+    profile = profileResolver.resolveProfile(clientScopes, Set.of("openid"));
+    assertThat(profile, is(iamProfile));
+
+    profile = profileResolver.resolveProfile(clientScopes, Set.of("openid", "wlcg", "aarc"));
+    assertThat(profile, is(iamProfile));
+
```

Comment:
based on the comment above, I would suggest the following test, if you agree with that outcome and it is not a backwards compatibility
```suggestion
    profile = profileResolver.resolveProfile(setOf("aarc", "wlcg"), Set.of("openid", "aarc", "kc"));
    assertThat(profile, is(aarcProfile));
```
(assuming "kc" is an actual profile)

Reply:
I'll design this with an expected Exception (kc is not part of client's scopes)
but I'll also add a test with 

```
profile = profileResolver.resolveProfile(setOf("aarc", "wlcg", "kc"), Set.of("aarc", "kc"));
assertThat(profile, is(defaultProfile));
```

But this can be part of a discussion as written above.

Reply:
Done